contracts/tvl-staking/RibbonDualStakingRewards.sol

```diff
@@ -0,0 +1,359 @@
+  uint256 public lastUpdateTime;
+
+  // timestamp dictating at what time of week to release rewards
+  // (ex: 1619226000 is Sat Apr 24 2021 01:00:00 GMT+0000 which will release as 1 am every saturday)
+  uint256 public startEmission;
+
+  mapping(address => Rewards) public userRewardPerTokenPaid;
+  mapping(address => Rewards) public rewards;
+
+  uint256 private _totalSupply;
+  mapping(address => uint256) private _balances;
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(
+    address _owner,
+    address _rewardsDistribution,
+    address _rewardsToken0,
+    address _rewardsToken1,
+    address _stakingToken,
+    uint256 _startEmission
+  ) Owned(_owner) {
+    require(_owner != address(0), "Owner must be non-zero address");
+    require(
+      _rewardsToken0 != address(0),
+      "Rewards token must be non-zero address"
+    );
+    require(
+      _rewardsToken1 != address(0),
+      "Rewards token must be non-zero address"
+    );
+    require(
+      _stakingToken != address(0),
+      "Staking token must be non-zero address"
+    );
+    require(
+      _rewardsDistribution != address(0),
+      "Rewards Distributor must be non-zero address"
+    );
+    require(
+      _startEmission > block.timestamp,
+      "Start Emission must be in the future"
+    );
+    require(
+      _rewardsToken0 != _stakingToken && _rewardsToken1 != _stakingToken,
+      "Rewards and staking tokens must be different"
+    );
+    require(
+      _rewardsToken0 != _rewardsToken1,
+      "Rewards tokens must be different"
+    );
+
+    rewardsToken0 = IERC20(_rewardsToken0);
+    rewardsToken1 = IERC20(_rewardsToken1);
+    stakingToken = IERC20(_stakingToken);
+    rewardsDistribution = _rewardsDistribution;
+    startEmission = _startEmission;
+  }
+
+  /* ========== VIEWS ========== */
+
+  function totalSupply() external view override returns (uint256) {
+    return _totalSupply;
+  }
+
+  function balanceOf(address account) external view override returns (uint256) {
+    return _balances[account];
+  }
+
+  // The minimum between periodFinish and the last instance of the current startEmission release time
+  function lastTimeRewardApplicable() public view override returns (uint256) {
+    return
+      Math.min(
+        _numWeeksPassed(block.timestamp).mul(1 weeks).add(startEmission),
+        periodFinish
+      );
+  }
+
+  function rewardPerToken() public view override returns (uint256, uint256) {
+    Rewards memory _rewardPerTokenStored = rewardPerTokenStored;
+    uint256 totalSupply_ = _totalSupply;
+    if (totalSupply_ == 0) {
+      return (_rewardPerTokenStored.token0, _rewardPerTokenStored.token1);
+    }
+
+    Rewards memory _rewardRate = rewardRate;
+    uint256 _lastTimeRewardApplicable = lastTimeRewardApplicable().sub(
+      lastUpdateTime
+    );
+    return (
+      uint256(_rewardPerTokenStored.token0).add(
+        _lastTimeRewardApplicable.mul(_rewardRate.token0).mul(1e18).div(
```

Comment:
if you listen to comment about [different duration of emissions](https://github.com/ribbon-finance/governance/pull/46/files#r773426339), then need to do `_lastTimeRewardApplicable.token0` instead